core/web/eth_keys_controller.go

```diff
@@ -379,3 +410,25 @@ func (ekc *ETHKeysController) setKeyMaxGasPriceWei(state ethkey.State, keyAddres
 	}
 	return presenters.SetETHKeyMaxGasPriceWei(utils.NewBig(price.ToInt()))
 }
+
+// getChainAndErr is a convenience wrapper to retrieve a chain for a given request
+// and call the corresponding API response error function for 400, 404 and 500 results
+func (ekc *ETHKeysController) getChainAndErr(c *gin.Context, cs evm.ChainSet, chainIDstr string) (chain evm.Chain, err1 error) {
```

Comment:
It's a little unusual to handle/respond as well as return an error, but if the error is unused, then an `ok bool` would be idiomatic:
```suggestion
func (ekc *ETHKeysController) getChain(c *gin.Context, cs evm.ChainSet, chainIDstr string) (chain evm.Chain, ok bool) {
```

Reply:
Thanks! I'll do that